### PR TITLE
[WIP] Rewrite Algs

### PR DESCRIPTION
**Pull request by rwoll**

### CODE REVIEW

File: paths.tex [hunk 1]
```diff
@@ -333,33 +333,38 @@ \section{ Shortest Path Algorithms}
...
-\end{lstlisting}
-\caption{An algorithm to compute the lengths of all consecutive temporal shortest
-paths}
+\begin{algorithm}[H]
+\SetAlgoLined
+\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest consecutive path)
```
Comment (ericthewry) [inline paths.tex]: Stay consistent with `src` vs `v_{src}`. Same for `dest` and `v_{dest}`.

Comment (rwoll) [inline paths.tex]: Ok -- I was treating the subscripts as indices totes to resolve types.

Comment (ericthewry) [inline paths.tex]: these need to be quantified. Whats the domain of `t`? `src` (`v_{src}`)? etc.

Also, need to stay consistent with how you use `t` vs. `t_i`.   Either subscript everywhere for this variable, or subscript nowhere.

File: paths.tex [hunk 2]
```diff
@@ -333,33 +333,38 @@ \section{ Shortest Path Algorithms}
...
+    start node $(t, v_{src})$ in $V \times \mathcal{T}$
+  }{
+    Let $e_{start} = (v_{src}, v_{src})$\;
+    Let shortest interval be $T(e_{start}) = (t, t)$\;
+    Let the reached edges min-heap $R = \{e_{start}\}$\;
+    Let the searched edges $S = \varnothing$\;
+    Let $dist[t][src][src] = 0$\;
+    \While{$R \neq \varnothing$}{
+      Let $e_{pred} = pop(R)$\;
+      \ForEach{$e_{succ} = (u,v) \in N_{e_{pred}}(v) - \{e_{succ}\} - S \cup R$}{
```
Comment (ericthewry) [inline paths.tex]: The notation for the edge neighborhood of a vertex `v` is `N_E(v)`, or, the neighborhood of `v` in the set `E`.  

Cant subtract `\{e_{succ}\}` because we dont know what it is. If you meant `\{e_pred\}` then okay. Ideally we'd really use `peek` instead of `pop` and then on 361 write `Insert pop(R) into S`, or insert `e_pred` into `S` at the beginning of the for-loop. But this is fine as is.

File: paths.tex [hunk 3]
```diff
@@ -333,33 +333,38 @@ \section{ Shortest Path Algorithms}
...
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest consecutive path)
+  distance from node $src$ to node $dest$ such that time $t_i$ intersects the time
+  interval of the first edge in the path.}
+  \ForEach(//loop over nodes in time slices){
+    start node $(t, v_{src})$ in $V \times \mathcal{T}$
+  }{
+    Let $e_{start} = (v_{src}, v_{src})$\;
```
Comment (ericthewry) [inline paths.tex]: Why not `e_{src}` for consistency and clarity?

File: paths.tex [hunk 4]
```diff
@@ -333,33 +333,38 @@ \section{ Shortest Path Algorithms}
...
+    Let shortest interval be $T(e_{start}) = (t, t)$\;
+    Let the reached edges min-heap $R = \{e_{start}\}$\;
+    Let the searched edges $S = \varnothing$\;
+    Let $dist[t][src][src] = 0$\;
+    \While{$R \neq \varnothing$}{
+      Let $e_{pred} = pop(R)$\;
+      \ForEach{$e_{succ} = (u,v) \in N_{e_{pred}}(v) - \{e_{succ}\} - S \cup R$}{
+        \If{$T(e_{pred}) \cap T(e_{succ}) \neq \varnothing$}{
+          Let $dist[t][src][v] = min(dist[t][src][u] + 1, dist[t][src][v])$\;
```
Comment (ericthewry) [inline paths.tex]: again, be consistent with `v_{src}` and `src`

File: paths.tex [hunk 5]
```diff
@@ -368,31 +373,40 @@ \subsection{Consecutive Shortest Path Distance Algorithm}
...
-\caption{An algorithm to compute the lengths of all complete temporal shortest
-paths}
+\begin{algorithm}[H]
+\SetAlgoLined
+\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest complete path)
+  distance from node $src$ to node $dest$ such that time $t_i$ intersects the time
```
Comment (ericthewry) [inline paths.tex]: Same as above

File: paths.tex [hunk 6]
```diff
@@ -333,33 +333,38 @@ \section{ Shortest Path Algorithms}
...
+\SetAlgoLined
+\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest consecutive path)
+  distance from node $src$ to node $dest$ such that time $t_i$ intersects the time
+  interval of the first edge in the path.}
+  \ForEach(//loop over nodes in time slices){
+    start node $(t, v_{src})$ in $V \times \mathcal{T}$
```
Comment (ericthewry) [inline paths.tex]: `time, start node pair`

File: paths.tex [hunk 7]
```diff
@@ -368,31 +373,40 @@ \subsection{Consecutive Shortest Path Distance Algorithm}
...
+\SetAlgoLined
+\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest complete path)
+  distance from node $src$ to node $dest$ such that time $t_i$ intersects the time
+  interval of the first edge in the path.}
+  \ForEach(//loop over nodes in time slices){
+    start node $(t, v_{src})$ in $V \times \mathcal{T}$
```
Comment (ericthewry) [inline paths.tex]: same as above

File: paths.tex [hunk 8]
```diff
@@ -368,31 +373,40 @@ \subsection{Consecutive Shortest Path Distance Algorithm}
...
+  }{
+    Let the reached edges min-heap $R = \{(v_{src}, v_{src})\}$\;
+    Let the searched edges $S = \varnothing$\;
+    Let $dist[t][src][src] = 0$\;
+    \For{$i$ in $range(|V|)$}{
+      Let $interval[t][i] = null$\;
```
Comment (ericthewry) [inline paths.tex]: can also pass `interval` as a parameter with the precondition that it is a complete map with defaults at null, removing the ugly (and not algorithmically interesting) for-loop.  Doing this seems more of an implementation detail than a algorithm detail

File: paths.tex [hunk 9]
```diff
@@ -368,31 +373,40 @@ \subsection{Consecutive Shortest Path Distance Algorithm}
...
+    }
+    \While{$R \neq \varnothing$}{
+      Let $e_{pred} = pop(R)$\;
+      \ForEach{$e_{succ} = (u,v) \in N_{e_{pred}}(v) - \{e_{succ}\} - S \cup R$}{
```
Comment (ericthewry) [inline paths.tex]: Same as above

File: paths.tex [hunk 10]
```diff
@@ -368,31 +373,40 @@ \subsection{Consecutive Shortest Path Distance Algorithm}
...
+        \If{$(\{t\}\cap(interval[t][e_{pred}] \cap T(e_{succ}))) \neq \varnothing$}{
+          Let $dist[t][src][v] = min(dist[t][src][u] + 1, dist[t][src][v])$\;
+          Let $interval[t][e_{succ}] = interval[t][e_{pred}] \cap T(e_{succ})$\;
+          Insert $e_{succ}$ into $R$\;
+        }
+      }
+      Insert $e_{succ}$ into $S$\;
```
Comment (ericthewry) [inline paths.tex]: This is outside of the scope for `e_{succ}`, should be inserting `e_{pred}`

File: paths.tex [hunk 11]
```diff
@@ -333,33 +333,38 @@ \section{ Shortest Path Algorithms}
...
-paths}
+\begin{algorithm}[H]
+\SetAlgoLined
+\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest consecutive path)
+  distance from node $src$ to node $dest$ such that time $t_i$ intersects the time
+  interval of the first edge in the path.}
```
Comment (ericthewry) [inline paths.tex]: Again, both in this description and below, I am a fan of calling them complete/total mappings with default values of `\infty`. Since we're not dealing with the i/o bound, there's no need to distinguish it as a 3D array.

**Comment by ericthewry** [comment]
@rwoll Left some comments. Lmk when those are fixed and I'll take another look.

**Comment by rwoll** [comment]
@ericthewry Thanks for the feedback! Before I start making changes can we make sure we are on the same page about accuracy of all the algs?
